metrics/system_tag.go

```diff
@@ -153,3 +153,91 @@ func (i *SystemTagSet) UnmarshalText(data []byte) error {
 	}
 	return nil
 }
+
+// NullSystemTagSet is a wrapper around SystemTagSet like guregu/null
+type NullSystemTagSet struct {
+	Set   *SystemTagSet
+	Valid bool
+}
```

Comment:
In principle:
Do we need the nullable type at this level? It resolves a config/option's issue so I would prefer if this new type remains a solution for those spaces and let the rest of the code only use SystemTagSet.
In this way, the rest of the code is independent of what we need on the config side. If in the future we change the way we configure k6, and use other types for the Null problem then all the code will be impacted. Instead, if it only relies on pure SystemTagSet then we don't have to change it.

In concrete, I may miss something marking my suggestion as not feasible. @joanlopez What do you think?